File: 2. Algorithmic Toolbox in GO/WEEK 3/5_collecting_signatures/covering_segments.go
package main

import "fmt"

func main() {
	var n , count int
	var result []int 
	fmt.Scan(&n)
	points := input(n)

	for len(points) > 0 {
		r_m := minima(points)
		points_copy := make([][]int , len(points))
		copy(points_copy , points)
		for 
	}
}

func minima(points [][]int) int {
	mini := points[0][1]
	for _ , i := range points {
		if mini > i[1] {
			mini = i[1]
		}
	}
	return mini
}

func input(n int) [][]int {
	points := make([][]int , n)
	for i := 0 ; i < n ; i++ {
		points[i] = make([]int , 2)
		for j := 0 ; j < 2 ; j++ {
			fmt.Scan(&points[i][j]) 
		}
	}
	return points
}

File: 2. Algorithmic Toolbox in GO/WEEK 3/5_collecting_signatures/covering_segments.py
points = []
for _ in range(int(input())):
    points.append(list(map(int,input().split())))

count , dots = 0 , []

while len(points) > 0:
    i = min(points , key= lambda x : x[1])[1]
    z = points[:]
    for p in z :
        if p[0] <= i <= p[1]:
            points.remove(p)
    if len(z) != len(points):
        count += 1
        dots.append(i)
print(count)
print(*dots)